services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowCrudSpec.groovy

```diff
@@ -151,6 +155,95 @@ class FlowCrudSpec extends BaseSpecification {
 
         where: flow << getSingleSwitchSinglePortFlows()
     }
+
+    def "Unable to create single-switch flow with the same ports and vlans on both sides"() {
+        given: "Potential single-switch flow with the same ports and vlans on both sides"
+        def flow = flowHelper.singleSwitchSinglePortFlow(topology.activeSwitches.first())
+        flow.destination.vlanId = flow.source.vlanId
+
+        when: "Try creating such flow"
+        northbound.addFlow(flow)
+
+        then: "Error is returned, stating a readable reason"
+        def error = thrown(HttpClientErrorException)
+        error.statusCode == HttpStatus.BAD_REQUEST
+        error.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: It is not allowed to create one-switch flow for the same ports and vlans"
```

Comment:
Is there sense to move such kind of messages to special error message constants? In the future, we may have lots of such kind of tests.

Reply:
I understand what you mean, but my thought process was that it is very unlikely that we will ever use this text in any other test and I didn't want to spread-out parts of the test to different places (away from the test/spec itself). We can move this and others to some Errors class that will contain all errors and I'm not too hell-bent to keeping current implementation. Do you want to change this?

Reply:
Yes, you are right that it is very unlikely that we will ever use this text in any other test again. But when such messages are located in some single place, it is very simple to make the corresponding changes if response messages are changed. Also, it is easy to understand what test cases are covered, what not, when we have all messages at a single place.

Reply:
1. Since we agreed that this will never be reused, then when response changes in both cases we will have to change only in 1 place, regardless.
2. There is no global Errors class with constants at the dev code to compare with our class, so I don't really understand how can we asses coverage by looking at Errors class. We do have specs and features that represent our test scenarios and coverage. Can you elaborate on how having errors in one place helps with understanding coverage? Do you mean that it will be like a short summary of what negative tests we have? This may be misleading / not representative

fyi, I already agree to move this to a separate class, just debating.
Can I/we do this in a separate patch? I guess this will have to do with refactoring and reviewing all existing tests to make it consistent. I can make an issue for it

Reply:
Yes, I meant a short summary of what negative tests we have. Ok, let's do it in a separate patch. It will be better.

Reply:
created #1622